pkg/resourcecollector/resourcetransformation.go

```diff
@@ -0,0 +1,138 @@
+package resourcecollector
+
+import (
+	"fmt"
+	"strings"
+
+	stork_api "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	storkops "github.com/portworx/sched-ops/k8s/stork"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Since we collect all resources from required migration namespace at once
+// getResourcePatch creates map of namespace: {kind: []resourceinfo{}}
+// to get transform spec for matching resources
+func GetResourcePatch(transformName string, namespaces []string) (map[string]stork_api.KindResourceTransform, error) {
+	// namespace- Kind:TransformSpec map for faster lookup
+	patch := make(map[string]stork_api.KindResourceTransform)
+	if transformName == "" {
+		logrus.Infof("empty name ")
+		return patch, nil
+	}
+	for _, namespace := range namespaces {
+		resp, err := storkops.Instance().GetResourceTransformation(transformName, namespace)
+		if err != nil && !errors.IsNotFound(err) {
+			logrus.Infof("not found in namespace: %v", transformName)
+			return nil, err
+		}
+		resMap := make(map[string][]stork_api.TransformResourceInfo)
+		for _, resource := range resp.Status.Resources {
```

Comment:
Pre-determining which resources to transform when we are validating Transform CR won't work if new objects get added to the namespaces which are getting migrated. I think we should simply validate the CR and check if the input arguments are correct. Once the migration triggers we could do this:
- collect all the objects
- dry run apply all the objects using the transformation
- fail or proceed the migration based on the dry run

Reply:
thats good point. Now we are deleting resource and setting transform CR status to `Initial` that way it will re-run all checks before we start migration